src/main/java/io/passport/server/controller/SurveyController.java

```diff
@@ -106,11 +156,21 @@ public ResponseEntity<?> updateSurvey(@PathVariable Long surveyId, @RequestBody
     /**
      * Delete by survey ID.
      * @param surveyId ID of the survey that is to be deleted.
+     * @param principal KeycloakPrincipal object that holds access token
      * @return
      */
     @DeleteMapping("/{surveyId}")
-    public ResponseEntity<?> deleteSurvey(@PathVariable Long surveyId) {
+    public ResponseEntity<?> deleteSurvey(@PathVariable Long surveyId,
+                                          @AuthenticationPrincipal KeycloakPrincipal<?> principal) {
         try{
+
+            // Allowed roles for this endpoint
+            List<Role> allowedRoles = List.of(Role.SURVEY_MANAGER);
```

Comment:
Instead of defining an allowedRoles variable in each method, can we just define them as private variables ? I think this will help us to get rid of duplicate variables.